cli/cmd/generate_execute.go

```diff
@@ -0,0 +1,392 @@
+		case id == 1 && !previewShown:
+			cli.OutputHuman("Resource details: \n")
+			for _, c := range data.plan.ResourceChanges {
+				cli.OutputHuman(fmt.Sprintf("  %s.%s will be %s\n", c.Type, c.Name,
+					createOrDestroy(
+						c.Change.Actions.Create(),
+						c.Change.Actions.Delete(),
+						c.Change.Actions.Update(),
+						c.Change.Actions.Read(),
+						c.Change.Actions.NoOp(),
+						c.Change.Actions.Replace(),
+						c.Change.Actions.CreateBeforeDestroy(),
+						c.Change.Actions.DestroyBeforeCreate(),
+					),
+				),
+				)
+			}
+			cli.OutputHuman("\n")
+			cli.OutputHuman("More details can be viewed by running:\n\n  cd %s\n  %s show tfplan.json\n", tf.WorkingDir(), tf.ExecPath())
+			cli.OutputHuman("\n")
+		default:
+			answer = id
+			prompt = false
+		}
+
+		if id == 1 && !previewShown {
+			previewShown = true
+		}
+	}
+
+	// Run apply
+	if answer == 0 {
+		return true, nil
+	}
+
+	// Quit
+	return false, nil
+}
+
+func processTfPlanChangesSummary(tf *tfexec.Terraform) (*tfPlanChangesSummary, error) {
+	// Extract changes from tf plan
+	cli.StartProgress("Getting terraform plan details")
+	plan, err := tf.ShowPlanFile(context.Background(), "tfplan.json")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to inspect terraform plan")
+	}
+	cli.StopProgress()
+
+	// Build output of changes
+	resourceCreate := 0
+	resourceDelete := 0
+	resourceUpdate := 0
+	resourceReplace := 0
+
+	for _, c := range plan.ResourceChanges {
+		switch {
+		case c.Change.Actions.Create():
+			resourceCreate += 1
+		case c.Change.Actions.Delete():
+			resourceDelete += 1
+		case c.Change.Actions.Update():
+			resourceUpdate += 1
+		case c.Change.Actions.Replace():
+			resourceReplace += 1
+		}
+	}
+
+	return &tfPlanChangesSummary{
+		plan:    plan,
+		create:  resourceCreate,
+		deleted: resourceDelete,
+		update:  resourceDelete,
+		replace: resourceReplace,
+	}, nil
+}
+
+func TerraformInit(tf *tfexec.Terraform) error {
+	cli.StartProgress("Running terraform init")
+	if err := tf.Init(context.Background(), tfexec.Upgrade(true)); err != nil {
+		return errors.Wrap(err, "failed to execute terraform init")
+	}
+	cli.StopProgress()
+
+	return nil
+}
+
+// Run terraform plan using the workingDir from *tfexec.Terraform
+//
+// - Run plan
+// - Get plan file details (returned)
+func TerraformExecPlan(tf *tfexec.Terraform) (*tfPlanChangesSummary, error) {
+	// Plan
+	cli.StartProgress("Running terraform plan")
+	if _, err := tf.Plan(context.Background(), tfexec.Out("tfplan.json")); err != nil {
+		return nil, err
+	}
+	cli.StopProgress()
+
+	// Gather changes from plan
+	return processTfPlanChangesSummary(tf)
+}
+
+// Run terraform apply using the workingDir from *tfexec.Terraform
+//
+// - Run plan
+// - Get plan file details (returned)
+func TerraformExecApply(tf *tfexec.Terraform) error {
+	// Plan
+	cli.StartProgress("Running terraform apply")
+	if err := tf.Apply(context.Background()); err != nil {
+		return err
+	}
+	cli.StopProgress()
```

Comment:
I believe you need to defer the stopProgress here. If line 282 gets hit the progress spinner won't get stopped.

Reply:
👍

Reply:
Reworked these to not use the shorthand if, but rather just define `err` and allow to run the start/stop around that line.